### PR TITLE
FEAT: make ScriptApproval a GlobalConfiguration for CasC.

### PR DESCRIPTION
**Pull request by dgarzon**
This PR enables Jenkins `scriptApproval.approvedSignatures` to be configured with CasC by following a different approach than https://github.com/jenkinsci/script-security-plugin/pull/219. 

Given we don't want to have glue code in this plugin, we need to make some modifications to `ScriptApproval` so that CasC can understand it. 

With this change users of CasC can now define the configuration as follows:
```
---
unclassified:
  scriptApproval:
    approvedSignatures:
    - method java.net.URI getHost
    - method java.net.URI getPort
    - new java.net.URI java.lang.String
```

### CODE REVIEW

**Comment by dgarzon** [comment]
@abayer you mentioned in the other PR that you would be willing to consider changes that make it easier for JCasC to interact with `script-security`, so long as there are no compatibility breaks. Can you provide some guidance as to wether my approach satisfies your requirements?

**Comment by abayer** [comment]
Worth considering - adding some reviewers besides myself.

**Comment by dgarzon** [comment]
Currently, all tests are passing, and I manually verified that this solution works with CasC. The one thing that I am not too happy with, is the fact that `ScriptApproval` still needs to implement `RootAction`. I would have liked to have a hidden `GlobalConfiguration` which could only be accessed by a `ScriptApprovalLink`, but could not find a way to do it. Any ideas?

**Comment by timja** [comment]
@abayer Would you mind reviewing this when you get a chance please?

**Comment by JeanMertz** [comment]
Would love to see this land 👍  This helps us move towards using configuration-as-code completely.

**Comment by er1c** [comment]
I have this same issue and just ran into this PR, would love to see it get added!

**Comment by timja** [comment]
@abayer or @jglick  would you mind taking a look please?
People keep asking about this on gitter

**Comment by michaelarichard** [comment]
I have this same issue and just ran into this PR, would love to see it get added as well!

File: src/main/java/org/jenkinsci/plugins/scriptsecurity/scripts/ScriptApproval.java [hunk 1]
```diff
@@ -688,39 +684,10 @@ public ApprovedWhitelist() {
...
-        }
-    }
-
-    @Override public synchronized void save() throws IOException {
-        final XmlFile configFile = getConfigFile();
-        if (configFile == null) {
-            throw new IOException("Cannot get config file. Probably, Jenkins is not ready");
-        }
-        configFile.write(this);
-        // TBD: outside synch block: SaveableListener.fireOnChange(this, getConfigFile());
```
[the remark this answers is not in the document]
Comment (jglick) [reply src/main/java/org/jenkinsci/plugins/scriptsecurity/scripts/ScriptApproval.java]: > there are some existing calls to `save` here that would have previously thrown exceptions during save up to their caller but would now just log an error and return successfully

The only nit I would find from this change is that those methods are still declared `throws IOException`, which is no longer actually true.

**Comment by casz** [comment]
@dgarzon do you have time to work on this, or would you like me to add a test-scoped dep to test JCasC is working as designed?

**Comment by placydo** [comment]
Any info when/if that will be merged?

**Comment by dgarzon** [comment]
@casz I will try to add the missing tests tomorrow 👍

**Comment by rafaljanicki** [comment]
Hey Guys,

Just a little check-in on that issue as it'd be great to have it.

Cheers

**Comment by asquel** [comment]
That would be awesome if somebody could approve this one :)

**Comment by timja** [comment]
@asquel its waiting on tests and conflicts being resolved,
@dgarzon are you wanting to continue working on this or should someone else finish it up?

**Comment by pjanouse** [comment]
We've used it for awhile and we didn't see any isses with it.

**Comment by pjanouse** [comment]
@dgarzon I've fixed the merge conflict at https://github.com/jenkinsci/script-security-plugin/compare/master...pjanouse:jcasc?expand=1, feel free to take it if you want.

**Comment by timja** [comment]
@pjanouse are you able to add the missing tests?
https://github.com/jenkinsci/configuration-as-code-plugin/blob/master/docs/PLUGINS.md#configuration-test

**Comment by pjanouse** [comment]
@timja working on it...

**Comment by pjanouse** [comment]
@timja Done in my branch (https://github.com/jenkinsci/script-security-plugin/compare/master...pjanouse:jcasc?expand=1)

**Comment by timja** [comment]
could you open a PR please @pjanouse

**Comment by pjanouse** [comment]
PR opened as #250

**Comment by olivergondza** [comment]
Closing in favor of #250

**Comment by akostadinov** [comment]
For anybody searching current working configuration is under `security`. See [here](https://github.com/jenkinsci/script-security-plugin/pull/250/files#diff-cebaf42ace8dcbbcb19447a8ef1435fa).

```
security:
  scriptApproval:
    approvedSignatures:
    - method java.net.URI getHost
```

And it works for me on Jenkins 2.190, casc plugin 1.36 and Script Security plugin 1.68.

**Pull request closed without merging**